src/realm/object-store/impl/apple/keychain_helper.cpp

```diff
@@ -78,20 +82,20 @@ CFPtr<CFMutableDictionaryRef> build_search_dictionary(CFStringRef account, CFStr
 }
 
 /// Get the encryption key for a given service, returning true if it either exists or the keychain is not usable.
-bool get_key(CFStringRef account, CFStringRef service, std::string_view group,
-             std::optional<std::vector<char>>& result, bool result_on_error = true)
+static bool get_key(CFStringRef account, CFStringRef service, std::string_view group,
+                    std::optional<EncryptionKey>& result, bool result_on_error = true)
 {
     auto search_dictionary = build_search_dictionary(account, service, string_view_to_cfstring(group).get());
     CFDataRef retained_key_data;
     switch (OSStatus status = SecItemCopyMatching(search_dictionary.get(), (CFTypeRef*)&retained_key_data)) {
         case errSecSuccess: {
             // Key was previously stored. Extract it.
             CFPtr<CFDataRef> key_data = adoptCF(retained_key_data);
-            if (key_size != CFDataGetLength(key_data.get()))
+            if (EncryptionKeySize != CFDataGetLength(key_data.get()))
                 return false;
 
-            auto key_bytes = reinterpret_cast<const char*>(CFDataGetBytePtr(key_data.get()));
-            result.emplace(key_bytes, key_bytes + key_size);
+            auto key_bytes = reinterpret_cast<const EncryptionKeyStorageType*>(CFDataGetBytePtr(key_data.get()));
```

Comment:
This is not a legal cast.

Reply:
That’s my fault, I’m very fond of this hack. One day the layout of std::array will change and I will be heartbroken. This should be manually copying and emplacing an std::array, but I forgot to remove my shortcut. Your comment about initializing the encryption key with an existing buffer applies here too.